tests/app/celery/test_service_callback_tasks.py

```diff
@@ -556,3 +582,71 @@ def test_create_delivery_status_callback_data_v3(
     assert data['status_reason'] == notification.status_reason
     assert data['provider'] == notification.sent_by
     assert data['provider_payload'] is None
+
+
+def test_check_and_queue_notification_callback_task_queues_task_with_proper_data(mocker, sample_notification):
+    test_url = 'https://test_url.com'
+    notification = sample_notification(callback_url=test_url)
+
+    notification_data = {'callback_url': test_url}
+    callback_signature_value = '6842b32e800372de4079e20d6e7e753bad182e44f7f3e19a46fd8509889a0014'
+
+    mocker.patch(
+        'app.celery.service_callback_tasks.create_delivery_status_callback_data_v3',
+        return_value=notification_data,
+    )
+    mocker.patch(
+        'app.celery.service_callback_tasks.generate_callback_signature',
+        return_value='6842b32e800372de4079e20d6e7e753bad182e44f7f3e19a46fd8509889a0014',
+    )
+    mock_delivery_status_from_notification = mocker.patch(
+        'app.celery.service_callback_tasks.send_delivery_status_from_notification.apply_async'
+    )
+
+    check_and_queue_notification_callback_task(notification)
+
+    # mock_create_callback_data.assert_called_once_with(notification, None, {})
```

Comment:
I'm assuming you meant to delete this comments.

Reply:
yes, good catch.